libminifi/src/core/yaml/YamlConfiguration.cpp

```diff
@@ -417,29 +405,27 @@ void YamlConfiguration::parseConnectionYaml(
 
 
         // Configure connection source
-        auto rawRelationship = connectionNode["source relationship name"]
-            .as<std::string>();
+        checkRequiredField(&connectionNode, "source relationship name", CONFIG_YAML_CONNECTIONS_KEY);
+        auto rawRelationship = connectionNode["source relationship name"].as<std::string>();
         core::Relationship relationship(rawRelationship, "");
-        logger_->log_debug(
-            "parseConnection: relationship => [%s]", rawRelationship);
+        logger_->log_debug("parseConnection: relationship => [%s]", rawRelationship);
         if (connection) {
           connection->setRelationship(relationship);
         }
 
         uuid_t srcUUID;
-        std::string connectionSrcProcName = connectionNode["source name"]
-            .as<std::string>();
+
         if (connectionNode["source id"]) {
-          std::string connectionSrcProcId = connectionNode["source id"]
-              .as<std::string>();
+          std::string connectionSrcProcId = connectionNode["source id"].as<std::string>();
 
           uuid_parse(connectionSrcProcId.c_str(), srcUUID);
         } else {
           // if we don't have a source id, try harder to resolve the source processor.
           // config schema v2 will make this unnecessary
+          checkRequiredField(&connectionNode, "source name", CONFIG_YAML_CONNECTIONS_KEY);
```

Comment:
@kevdoran I like that you documented your assumption. Could you augment that comment with a log message affirming the decision of the code so that we could better debug "how and why?"